context/on_startup.py

```diff
@@ -1,39 +1,26 @@
-from os import listdir
 import json
 import SimpleHTTPServer
 import SocketServer
-import re
-
-def as_track(file):
-    return {
-        "name": re.sub(r'.*\/', '', file),
-        "url": 'data/' + file,
-        "format": re.sub(r'.*\.', '', file)
-        # TODO: indexURL if BAM
-    }
 
-def populate_data_directory():
-    # files.txt is really for testing rather than production use.
-    files = '\n'.join(listdir('data'))
-    with open('data/files.txt', 'w') as list_file:
-        list_file.write(files)
 
-    # options.json is actually used by IGV.js.
-    # TODO: How does Refinery communicate the genome assembly?
+def populate_igv_configuration():
+    with open("data/input.json") as f:
+        config_data = json.loads(f.read())
+
+    tracks = []
     assembly = "hg19"
-    url_base = "https://s3.amazonaws.com/data.cloud.refinery-platform.org/data/igv-reference/{}/".format(assembly)
-
-    tracks = [as_track(f) for f in listdir('data') if f not in ["files.txt", "hello_world.txt"]]
-    tracks.append({
-        "name": "Genes",
-        "type": "annotation",
-        "format": "bed",
-        "sourceType": "file",
-        "url": url_base + "refGene.bed",
-        "indexURL": url_base + "refGene.bed.tbi",
-        "visibilityWindow": 300000000,
-        "displayMode": "EXPANDED"
-    })
+    url_base = "https://s3.amazonaws.com/data.cloud.refinery-platform.org/data/igv-reference/{}/".format(
+        assembly
+    )
+
+    for url in config_data["file_relationships"]:
+        tracks.append({
+            "name": url.split("/")[-1],
```

Comment:
Fine, but os.path.basename() would make your intention a little more clear. (Though this line will be changing soon, anyway, so not worth worrying about.)